packages/cli/src/utils/object.ts

```diff
@@ -0,0 +1,30 @@
+export type PlainObject = Record<string, any>;
+
+export function isPlainObject(value: unknown): value is PlainObject {
+  return Boolean(value) && typeof value === 'object' && !Array.isArray(value);
+}
+
+export function mergeDeep<T extends PlainObject>(target: T, source?: PlainObject): T {
```

Comment:
nitpick:

Normally, I'm more in favor of local solutions, but if we want to avoid the maintenance of this function, `defu` https://github.com/unjs/defu should be available through `c12` and is more battle-tested and performant. If we want to make sure we can explicitly install `defu` 

Otherwise, we should add a unit test for it at least.